tools/sync-module-dependencies/internal/generate-replaces.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/grafana/replace-generator/internal/helpers"
+	"github.com/grafana/replace-generator/internal/types"
+)
+
+func main() {
+	fileHelper := helpers.GetFileHelper()
+
+	projectReplaces, err := fileHelper.LoadProjectReplaces()
+	if err != nil {
+		log.Fatalf("Failed to load project replaces: %v", err)
+	}
+
+	normalizeComments(projectReplaces.Replaces)
+
+	// Group modules by file type and generate one output file per file type
+	fileTypesSeen := make(map[types.FileType]bool)
+	for _, module := range projectReplaces.Modules {
+		if !fileTypesSeen[module.FileType] {
+			if err := generateReplacesForFileType(fileHelper, projectReplaces, module.FileType); err != nil {
+				log.Fatalf("Failed to generate replaces for file type %q: %v", module.FileType, err)
+			}
+			fileTypesSeen[module.FileType] = true
+		}
+	}
+}
+
+// Removes unnecessary newlines and whitespaces from comment metadata
+func normalizeComments(entries []types.ReplaceEntry) {
+	for i := range entries {
+		entries[i].Comment = strings.ReplaceAll(entries[i].Comment, "\n", " ")
+		entries[i].Comment = strings.TrimSpace(entries[i].Comment)
+	}
+}
+
+// Generates the .txt files that contain the replace directives formatted into their corresponding template
```

Comment:
Not sure why do we need to save the .txt file on disk? Can't it be held in memory as a string which would make the code simpler?